kbl2vec/src/main/java/com/foursoft/harness/kbl2vec/transform/geometry/geo_2d/OccurrenceOrUsageViewItem2DTransformer.java

```diff
@@ -0,0 +1,85 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.kbl2vec.transform.geometry.geo_2d;
+
+import com.foursoft.harness.kbl.common.HasAliasId;
+import com.foursoft.harness.kbl.common.HasIdentification;
+import com.foursoft.harness.kbl.common.HasPlacement;
+import com.foursoft.harness.kbl.v25.ConnectionOrOccurrence;
+import com.foursoft.harness.kbl.v25.KblAliasIdentification;
+import com.foursoft.harness.kbl.v25.KblTransformation;
+import com.foursoft.harness.kbl2vec.core.Query;
+import com.foursoft.harness.kbl2vec.core.TransformationContext;
+import com.foursoft.harness.kbl2vec.core.TransformationResult;
+import com.foursoft.harness.kbl2vec.core.Transformer;
+import com.foursoft.harness.vec.common.util.StreamUtils;
+import com.foursoft.harness.vec.v2x.VecAliasIdentification;
+import com.foursoft.harness.vec.v2x.VecOccurrenceOrUsageViewItem2D;
+import com.foursoft.harness.vec.v2x.VecPartOccurrence;
+import com.foursoft.harness.vec.v2x.VecTransformation2D;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+public class OccurrenceOrUsageViewItem2DTransformer
+        implements Transformer<ConnectionOrOccurrence, VecOccurrenceOrUsageViewItem2D> {
+
+    @Override
+    public TransformationResult<VecOccurrenceOrUsageViewItem2D> transform(final TransformationContext context,
+                                                                          final ConnectionOrOccurrence source) {
+        if (source instanceof final HasPlacement<?> hasPlacement) {
+            final VecOccurrenceOrUsageViewItem2D destination = new VecOccurrenceOrUsageViewItem2D();
+
+            if (source instanceof final HasIdentification hasIdentification && StringUtils.isNotBlank(
+                    hasIdentification.getId())) {
+                destination.setIdentification(hasIdentification.getId());
+            } else {
+                context.getLogger().warn("Occurrence (xml ID: {}) has no ID.", source.getXmlId());
+                destination.setIdentification("ViewItem2D_" + source.getXmlId());
+            }
+
+            final TransformationResult.Builder<VecOccurrenceOrUsageViewItem2D> builder = TransformationResult.from(
+                    destination);
+
+            if (source instanceof final HasAliasId<?> hasAliasId) {
+                final List<KblAliasIdentification> aliasIds = hasAliasId.getAliasIds().stream()
+                        .flatMap(StreamUtils.ofClass(KblAliasIdentification.class))
+                        .toList();
+                builder.withDownstream(KblAliasIdentification.class, VecAliasIdentification.class,
+                                       Query.fromLists(aliasIds), VecOccurrenceOrUsageViewItem2D::getAliasIds);
+            }
+
+            final KblTransformation transformation = (KblTransformation) hasPlacement.getPlacement();
```

Comment:
I would say, if `getPlacement() == null` there should be now view since the item is obviously not placed some in the 2D (or 3D space). So I would think line 51 should be 
```java
if (source instanceof final HasPlacement<?> hasPlacement && hasPlacement.getPlacement() != null) {
```
But a second opinion would be good. @RainerGanss what do you think?

Reply:
I agree. I would not expect to find a ViewItem that is empty. If there is no placement in the KBL (placement == null) the creation of the viewItem (2D and 3D) should be skipped.